plate_mapper/plate_mapper.py

```diff
@@ -0,0 +1,91 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, The Wetlab Assistant Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+import argparse
+
+
+def plate_mapper(input_f, barseq_f, output_f):
+    """ Convert a plate map file into a mapping file
+
+    Parameters
+    ---------
+    input_f : file object
+        Input plate map file
+    barseq_f : file object
+        Barcode sequence template file
+    output_f : file object
+        Output mapping file
+    """
+    idx = ''  # primer plate ID
+    cols = 0  # number of columns
+    plates = {}  # plates (primer plate index : well ID : sample ID)
+    properties = {}  # properties
+    # Read input plate map file
+    print('Reading input plate map file...')
+    for line in input_f:
+        line = line.rstrip('\r\n')
+        if not line:  # skip empty lines
+            continue
+        if not ''.join(line.split()):
+            continue
+        l = line.split('\t')
+        if l[1] == '1':  # plate starts
+            if not cols:  # count number of columns
```

Comment:
Also you could you these 2 ifs to make it clearer, right?
`if l[1] == '1' and cols == 0:`